server/controllers/finance/reports/income_expense/index.js

```diff
@@ -35,98 +35,96 @@ const DEFAULT_PARAMS = {
 // expose to the API
 exports.document = document;
 
-function document(req, res, next) {
-  let report;
-
+async function document(req, res, next) {
   const options = _.defaults(req.query, DEFAULT_PARAMS);
 
   try {
-    report = new ReportManager(TEMPLATE, req.session, options);
+    const report = new ReportManager(TEMPLATE, req.session, options);
+    const { periodFromId, periodToId, fiscalYearId } = options;
+    const data = {};
+
+    const [periodFrom, periodTo, fiscalYear] = await Promise.all([
+      getPeriodById(periodFromId),
+      getPeriodById(periodToId),
+      Fiscal.lookupFiscalYear(fiscalYearId),
+    ]);
+
+    if (periodFrom.start_date > periodTo.start_date) {
```

Comment:
I wonder if this check could be incorporated into bh-fiscal-period-select (in another PR)?

Reply:
This is a good idea.  I see you made an issue #6233